package/kernel/linux/modules/crypto.mk

```diff
@@ -93,6 +130,69 @@ endef
 $(eval $(call KernelPackage,crypto-ccm))
 
 
+define KernelPackage/crypto-chacha20poly1305
+  TITLE:=ChaCha20 and Poly1305 CryptoAPI modules
+  KCONFIG:= \
+	CONFIG_CRYPTO_CHACHA20POLY1305 \
+	CONFIG_CRYPTO_CHACHA20_X86_64 \
+	CONFIG_CRYPTO_POLY1305_X86_64 \
+	CONFIG_CRYPTO_CHAHA20_NEON \
+	CONFIG_CRYPTO_POLY1305_NEON \
+	CONFIG_CRYPTO_POLY1305_ARM \
+	CONFIG_CRYPTO_CHACHA_MIPS \
+	CONFIG_CRYPTO_POLY1305_MIPS
+  FILES:= \
+  	$(LINUX_DIR)/crypto/chacha20poly1305.ko \
+	$(LINUX_DIR)/lib/crypto/libchacha.ko \
+	$(LINUX_DIR)/lib/crypto/libchacha20poly1305.ko
+  AUTOLOAD:=$(call AutoLoad,09,rfc7539)
+  $(call AddDepends/crypto, +kmod-crypto-aead @!LINUX_5_4)
+endef
+
+define KernelPackage/crypto-chacha20poly1305/x86/64
+  FILES+= \
+	$(LINUX_DIR)/arch/x86/crypto/poly1305-x86_64.ko \
+	$(LINUX_DIR)/arch/x86/crypto/chacha-x86_64.ko
+  AUTOLOAD+= \
+	$(call AutoLoad,09,poly1305-x86_64) \
+	$(call AutoLoad,09,chacha-x86_64)
+endef
+
+define KernelPackage/crypto-chacha20poly1305/arm
+  FILES+=$(LINUX_DIR)/arch/arm/crypto/poly1305-arm.ko
+  AUTOLOAD+=$(call AutoLoad,09,poly1305-arm)
+endef
+
+KernelPackage/crypto-chacha20poly1305/kirkwood=$(KernelPackage/crypto-chacha20poly1305/arm)
+
+define KernelPackage/crypto-chacha20poly1305/arm-neon
+  $(call KernelPackage/crypto-chacha20poly1305/arm)
+  FILES+= \
+	$(LINUX_DIR)/arch/arm/crypto/nhpoly1305-neon.ko \
+	$(LINUX_DIR)/arch/arm/crypto/chacha-neon.ko
+  AUTOLOAD+= \
+	$(call AutoLoad,09,nhpoly1305-neon) \
+	$(call AutoLoad,09,chacha-neon)
+endef
+
+KernelPackage/crypto-chacha20poly1305/imx6=$(KernelPackage/crypto-chacha20poly1305/arm-neon)
+KernelPackage/crypto-chacha20poly1305/ipq40xx=$(KernelPackage/crypto-chacha20poly1305/arm-neon)
+KernelPackage/crypto-chacha20poly1305/mvebu/cortexa9=$(KernelPackage/crypto-chacha20poly1305/arm-neon)
+
+define KernelPackage/crypto-chacha20poly1305/mips
+  FILES+= \
+	$(LINUX_DIR)/arch/mips/crypto/poly1305-mips.ko \
+	$(LINUX_DIR)/arch/mips/crypto/chacha-mips.ko
+  AUTOLOAD+= \
+	$(call AutoLoad,09,poly1305-mips) \
+	$(call AutoLoad,09,chacha-mips)
+endef
+
+KernelPackage/crypto-chacha20poly1305/ramips=$(KernelPackage/crypto-chacha20poly1305/mips)
```

Comment:
Hmm, shouldn't this be added to the other MIPS targets as well?

Reply:
Yeah, maybe, but I don't know which ones. There are probably a bunch of other ARM(-NEON) targets missing too. Trying to only enable what I can test or from examples elsewhere in this file.